src/chrono/io.cljc

```diff
@@ -6,74 +6,73 @@
             #?(:cljs [goog.string.format]))
   (:refer-clojure :exclude [format]))
 
-(defn- format-str [fmt val width]
-  (->>
-    (apply
-      #?(:clj  clojure.core/format
-         :cljs goog.string/format)
-      fmt
-      [val])
-    (take-last width)
-    (str/join)))
+(defn- format-str [v [fmt & fmt-args] lang]
+  (if (and (some? lang) (contains? (util/locale lang) fmt))
+    (let [full? (empty? (filter #(= % :short) fmt-args))]
+      (-> (util/locale lang)
+          fmt
+          (get-in [v (if full? :name :short)])))
 
+    (let [width (or (first (filter integer? fmt-args)) (util/format-patterns fmt))]
+      (if width
+        (->>
+         (apply
+          #?(:clj  clojure.core/format
```

Comment:
Better to get rid of this clj/cljs format and instead write it from scratch because it is used only for padding with zeroes

Reply:
Not sure about efficiency but resolved with 300fb17e9cb136ac43b86a7a69140570a4a89732 ea11921aae788c4df0ad9d374733b3f0fae834f4